jetty-xml/src/main/java/org/eclipse/jetty/xml/XmlParser.java

```diff
@@ -127,10 +131,10 @@ public boolean isValidating()
         return _parser.isValidating();
     }
 
-    public synchronized void redirectEntity(String name, URL entity)
+    public void redirectEntity(String name, URL entity)
     {
         if (entity != null)
-            _redirectMap.put(name, entity);
+            _lock.runLocked(() -> _redirectMap.put(name, entity));
```

Comment:
Since you have a getter for the lock and neither the class nor method is final then you should use the getter: 
```suggestion
            lock().runLocked(() -> _redirectMap.put(name, entity));
```